third_party/amd/lib/TritonAMDGPUToLLVM/SchedInstructions.cpp

```diff
@@ -497,6 +497,19 @@ struct TritonAMDGPULowerInstructionSchedHints
   }
 };
 
+static bool hasDotOpAndExpOp(scf::ForOp forOp) {
```

Comment:
This is a very loose check. I think we can extract the `isChainDotHead` and `isChainDotTail` in accelerate matmul pass to a common utility file to avoid recreating the logic here.